tests/test_builtin_traits.py

```diff
@@ -100,3 +101,28 @@ def test_has_parent_verify():
 
     op = Parent2Op(regions=[[HasMultipleParentOp()]])
     op.verify()
+
+
+@irdl_op_definition
+class IsTerminatorOp(IRDLOperation):
+    """
+    An operation that provides the IsTerminator trait.
+    """
+
+    name = "test.is_terminator"
+
+    successor: OptSuccessor
+
+    traits = frozenset([IsTerminator()])
+
+
```

Comment:
You should probably add a test that will fail if a terminator is not the last operation (after adding the check in the trait)